plugins/inputs/nomad/nomad_test.go

```diff
@@ -0,0 +1,127 @@
+package nomad
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNomadStats(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/v1/metrics" {
+			w.WriteHeader(http.StatusOK)
+			_, err := fmt.Fprintln(w, responseKeyMetrics)
+			require.NoError(t, err)
+		}
+	}))
+	defer ts.Close()
+
+	labelFilter, _ := filter.NewIncludeExcludeFilter([]string{"host"}, nil)
+
+	n := &Nomad{
+		URL:         ts.URL,
+		labelFilter: labelFilter,
+	}
+
+	var acc testutil.Accumulator
+	err := acc.GatherError(n.Gather)
+	require.NoError(t, err)
+
+	fields := map[string]interface{}{
+		"value": float32(500),
+	}
+	tags := map[string]string{
+		"node_scheduling_eligibility": "eligible",
+		"host":                        "node1",
+		"node_id":                     "2bbff078-8473-a9de-6c5e-42b4e053e12f",
+		"datacenter":                  "dc1",
+		"node_class":                  "none",
+		"node_status":                 "ready",
+	}
+	acc.AssertContainsTaggedFields(t, "nomad.client.allocated.cpu", fields, tags)
+
+	fields = map[string]interface{}{
+		"count": int(7),
+		"max":   float64(1),
+		"min":   float64(1),
+		"mean":  float64(1),
+		"rate":  float64(0.7),
+		"sum":   float64(7),
+		"sumsq": float64(0),
+	}
+	tags = map[string]string{
+		"host": "node1",
+	}
+	acc.AssertContainsTaggedFields(t, "nomad.nomad.rpc.query", fields, tags)
+
+	fields = map[string]interface{}{
+		"count": int(20),
+		"max":   float64(0.03747599944472313),
+		"min":   float64(0.003459000028669834),
+		"rate":  float64(0.026318199979141355),
+		"sum":   float64(0.26318199979141355),
+		"sumsq": float64(0),
+		"mean":  float64(0.013159099989570678),
+	}
+	tags = map[string]string{
+		"host": "node1",
+	}
+	acc.AssertContainsTaggedFields(t, "nomad.memberlist.gossip", fields, tags)
+
+}
+
+var responseKeyMetrics = `
```

Comment:
Please move this test-data to a file e.g. in `testdata/response_key_metrics.json` and load it on test. This improves readability and allows easy extension of tests.